### PR TITLE
More classification perf fixes.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Span change is to address this:

image

Have commented the rest.

### CODE REVIEW

File: src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs [hunk 1]
```diff
@@ -138,19 +123,18 @@ CandidateReason.Ambiguous or
...
+        SimpleNameSyntax name,
         [NotNullWhen(returnValue: true)] ISymbol? symbol,
         out ClassifiedSpan classifiedSpan)
     {
         // For Namespace parts, we want don't want to classify the QualifiedNameSyntax
         // nodes, we instead wait for the each IdentifierNameSyntax node to avoid
         // creating overlapping ClassifiedSpans.
-        if (symbol is INamespaceSymbol namespaceSymbol &&
-            name is IdentifierNameSyntax identifierNameSyntax)
+        if (symbol is INamespaceSymbol namespaceSymbol)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs:133]:
```suggestion
        if (symbol is INamespaceSymbol namespaceSymbol &&
            name is IdentifierNameSyntax)
```

File: src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs [hunk 2]
```diff
@@ -29,36 +29,21 @@ internal sealed class NameSyntaxClassifier : AbstractNameSyntaxClassifier
...
     }
 
     public override ImmutableArray<Type> SyntaxNodeTypes { get; } =
     [
-        typeof(AliasQualifiedNameSyntax),
-        typeof(GenericNameSyntax),
         typeof(IdentifierNameSyntax),
-        typeof(QualifiedNameSyntax),
-        typeof(SimpleNameSyntax),
+        typeof(GenericNameSyntax),
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs:39]: no need to check these other 'composed name types'.  We only care about `X` and `X<...>` (and we only classifiy the 'X' part of each).  Every other name type composes into one of these two name types that we actually classify here.

this addresses a *lot* of callbacks for dotted names that serve no purpose.

File: src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs [hunk 3]
```diff
@@ -29,36 +29,21 @@ internal sealed class NameSyntaxClassifier : AbstractNameSyntaxClassifier
...
     ];
 
-    protected override int? GetRightmostNameArity(SyntaxNode node)
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs:47]: unused method.

File: src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs [hunk 4]
```diff
@@ -293,18 +278,17 @@ private static bool IsInVarContext(NameSyntax name)
...
-        NameSyntax name,
+        SimpleNameSyntax name,
         SymbolInfo symbolInfo,
         SegmentedList<ClassifiedSpan> result)
     {
         // Okay - it wasn't a type. If the syntax matches "var q = from" or "q = from", and from
         // doesn't bind to anything then optimistically color from as a keyword.
-        if (name is IdentifierNameSyntax identifierName &&
-            identifierName.Identifier.HasMatchingText(SyntaxKind.FromKeyword) &&
+        if (name.Identifier.HasMatchingText(SyntaxKind.FromKeyword) &&
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs:287]:
```suggestion
        if (name if IdentifierNameSyntax &&
            name.Identifier.HasMatchingText(SyntaxKind.FromKeyword) &&
```

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**